core/arch/arm/kernel/tee_time_arm_cntpct.c

```diff
@@ -54,3 +56,39 @@ static const struct time_source arm_cntpct_time_source = {
 };
 
 REGISTER_TIME_SOURCE(arm_cntpct_time_source)
+
+/*
+ * We collect jitter using CNTPCT in 32- or 64-bit mode that is clocked at the
+ * CPU core rate.
```

Comment:
You said it once, it's true.  But you just waved your hand at it last time in a way I was unable to interpret.

The "lumpy increment" is news to me but it makes sense.  Where the interval is too large that in fact nothing was added to the counter, nobody can do anything, and your old implementation did nothing useful either.

However when the counter has changed, we can get entropy from it by XORing every 2-bit pair in the counter on to the 2-bit slot being updated in turn.

I also changed the accumulator to rotate, since we don't need to reset the two bits with all those XORs either.

So thanks for the tip ^^